lib/ansible/modules/storage/netapp/na_ontap_qos_adaptive_policy_group.py

```diff
@@ -0,0 +1,336 @@
+#!/usr/bin/python
+
+# (c) 2018-2019, NetApp, Inc
+# GNU General Public License v3.0+
+# (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'certified'}
+
+DOCUMENTATION = '''
+module: na_ontap_qos_adaptive_policy_group
+short_description: NetApp ONTAP Adaptive Quality of Service policy group.
+extends_documentation_fragment:
+  - netapp.na_ontap
+version_added: '2.9'
+author: NetApp Professional Services - (@joshedmonds) <edmonds@netapp.com>
+
+description:
+  - Create, destroy, modify, or rename an Adaptive QoS policy group on NetApp ONTAP. Module is based on the standard QoS policy group module.
+
+options:
+  state:
+    choices: ['present', 'absent']
+    description:
+      - Whether the specified policy group should exist or not.
+    default: 'present'
+    type: str
+
+  name:
+    description:
+    - The name of the policy group to manage.
+    type: str
```

Comment:
Indicate that this is a required parameter.  Same thing for server

Reply:
Done.